src/Common/Database/Models/Ledger/Substates/ValidatorSystemMetadataSubstate.cs

```diff
@@ -0,0 +1,176 @@
+ * password, loss of private key, unauthorised use or misuse of such passwords or keys;
+ *
+ * F. any losses including loss of anticipated savings or other benefits resulting from
+ * use of the Work or any changes to the Work (however implemented).
+ *
+ * You are solely responsible for; testing, validating and evaluation of all operation
+ * logic, functionality, security and appropriateness of using the Work for any commercial
+ * or non-commercial purpose and for any reproduction or redistribution by You of the
+ * Work. You assume all risks associated with Your use of the Work and the exercise of
+ * permissions under this License.
+ */
+
+using Common.Extensions;
+using Microsoft.EntityFrameworkCore;
+using System.ComponentModel.DataAnnotations.Schema;
+using System.Text;
+using Core = RadixCoreApi.Generated.Model;
+using Validator = Common.Database.Models.Ledger.Normalization.Validator;
+
+namespace Common.Database.Models.Ledger.Substates;
+
+/// <summary>
+/// Represents Validator fork votes.
+/// </summary>
+[Index(nameof(ValidatorId))]
+[Table("validator_system_metadata_substates")]
+public class ValidatorSystemMetadataSubstate : DataSubstateBase
+{
+    [Column(name: "validator_id")]
+    public long ValidatorId { get; set; }
+
+    [ForeignKey(nameof(ValidatorId))]
+    public Validator Validator { get; set; }
+
+    // These are all [Owned] types below - exactly one of these will be present on each
+    public ValidatorCandidateForkVote? ValidatorCandidateForkVote { get; set; }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ValidatorSystemMetadataSubstate"/> class.
+    /// The SubstateBase properties should be set separately.
+    /// </summary>
+    public ValidatorSystemMetadataSubstate(Validator validator, ValidatorCandidateForkVote validatorCandidateForkVote)
+    {
+        Validator = validator;
+        ValidatorCandidateForkVote = validatorCandidateForkVote;
+    }
+
+    private ValidatorSystemMetadataSubstate()
+    {
+    }
+
+    public bool SubstateMatches(ValidatorSystemMetadataSubstate otherSubstate)
+    {
+        return Validator == otherSubstate.Validator
+               && ValidatorCandidateForkVote == otherSubstate.ValidatorCandidateForkVote;
+    }
+}
+
+[Owned]
+// Aka ValidatorSystemMetadata in the engine
+public record ValidatorCandidateForkVote
+{
+    /// <summary>
+    ///  The FullBytes field is technically schema-less according to the engine, but a 32 byte vote will be
+    ///  interpreted as a vote from that validator for the given fork.
+    ///  We attempt to parse out the meaning from the FullBytes field into the other fields.
+    /// </summary>
+    [Column(name: "full_bytes")]
+    public byte[] FullBytes { get; set; }
+
+    [Column(name: "fork_name")]
+    public string? ForkName { get; set; }
+
+    [Column(name: "fork_id")]
+    public byte[]? ForkId { get; set; }
+
+    [Column(name: "nonce")]
+    public byte[]? Nonce { get; set; }
```

Comment:
It is a _hash_ of the nonce (and validator's public key), can we call it `NonceHash`? Also for consistency with the Java code.